### PR TITLE
feat(payment): PAYPAL-2616 moved braintree-paypal-payment-strategy to braintree-integration package

### PR DESCRIPTION
**Pull request by bc-nick**
## What?

Moved `braintree-paypal-payment-strategy` to `braintree-integration` package

## Why?

To use the new checkout payment integration JS API

@bigcommerce/team-checkout @bigcommerce/team-payments

### CODE REVIEW

**Review by bc-dronov — APPROVED** [review]

**Review by andriiVitvitskyi1990 — APPROVED** [review]

File: packages/braintree-utils/src/braintree-integration-service.ts [hunk 1]
```diff
@@ -17,10 +20,13 @@ import {
     BraintreeError,
     BraintreeHostWindow,
     BraintreeModule,
+    BraintreePaypal,
     BraintreePaypalCheckout,
     BraintreePaypalSdkCreatorConfig,
     BraintreeShippingAddressOverride,
     BraintreeThreeDSecure,
+    // BraintreeThreeDSecureOptions,
```
[the remark this answers is not in the document]
Comment (bc-nick) [reply packages/braintree-utils/src/braintree-integration-service.ts]: removed

File: packages/braintree-integration/src/braintree-paypal/braintree-paypal-payment-options.ts
[the remark this answers is not in the document]
Comment (bc-nick) [reply packages/braintree-integration/src/braintree-paypal/braintree-paypal-payment-options.ts:1]: Good point. Renamed not only this file but `braintree-paypal-customer-options.ts` to `braintree-paypal-customer-initialize-options.ts`. You can take a look

File: packages/braintree-utils/src/braintree.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import { Omit } from '@bigcommerce/checkout-sdk/payment-integration-api';
```
[the remark this answers is not in the document]
Comment (bc-nick) [reply packages/braintree-utils/src/braintree.ts]: Thanks! Removed

[the remark this answers is not in the document]
Comment (bc-nick) [reply packages/braintree-utils/src/braintree.ts:1]: I think I figured out why I imported this method from the `payment-integration-api`)

Screenshot 2023-12-05 at 12 40 04

**Review by serhii-tkachenko — APPROVED** [review]

**Pull request merged**